salt/modules/file.py

```diff
@@ -3213,6 +3213,62 @@ def touch(name, atime=None, mtime=None):
     return os.path.exists(name)
 
 
+def tail(path, lines):
+    '''
+    .. versionadded:: Neon
+
+    Read the last n lines from a file
+
+    path
+        path to file
+
+    lines
+        number of lines to read
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' file.tail /path/to/file 10
+    '''
+    path = os.path.expanduser(path)
+    lines_found = []
+
+    if not os.path.isfile(path):
+        raise SaltInvocationError('File not found: {0}'.format(path))
+
+    if not __utils__['files.is_text'](path):
+        raise SaltInvocationError(
+            'Cannot tail a binary file: {0}'.format(path))
+
+    try:
+        lines = int(lines)
+    except ValueError:
+        log.error('file.tail: \'lines\' value must be an integer')
+        lines = 10
+
+    try:
+        with salt.utils.fopen(path) as tail_fh:
+            tail_fh.seek(0, os.SEEK_END)
+            position = tail_fh.tell()
+            line = ''
+            while position >= 0:
+                tail_fh.seek(position)
+                next_char = tail_fh.read(1)
+                if next_char == os.linesep:
```

Comment:
This won't work on Windows because `os.linesep` is '\r\n' on that platform. I would suggest you use '\n' instead of `os.linesep`.

Reply:
I don't understand. If it's a Windows system, won't the file have `\r\n` in it? Why would you expect a Windows file to only have `\n` in it?

Reply:
`os.linesep` is 2 characters in Windows `\r` and `\n`. You're comparing a single character with 2 characters on Windows in this case.